### DISCUSSION TITLE
JS SDK v0.7.0: "An unknown error occured" on both devnet and testnet

### DISCUSSION THREAD

**Discussion by a3s7p** [NillionNetwork/discussions#154] [Bugs]
I have been trying to update [Wordle](https://github.com/a3s7p/wordle) to:

```typescript
"@nillion/client-react-hooks": "^0.2.0",
"@nillion/client-vms": "^0.2.0",
"@nillion/client-wasm": "^0.2.0",
```

However I am now getting this a couple of seconds after approving the transaction in Keplr:

<img width="1454" alt="image" src="https://github.com/user-attachments/assets/98cb0ec5-1a54-48b0-9bf9-d6f3291cdcfb">

The code where it occurs:

```typescript
const nilCompute = useNilInvokeCompute()

nilCompute.execute({
  programId: wordle.programId,
  inputBindings: [
    {party: "Player", user: client.id},
    {party: "Gamemaker", user: gmUid},
  ],
  outputBindings: [{party: "Player", users: [client.id]}],
  computeTimeValues: chars.map((char, i) => ({
    name: `guess_${i + 1}`,
    value: NadaValue.new_secret_integer(char.charCodeAt(0).toString()),
  })),
  valueIds: [wordle.gmStoreId],
})
```

Same thing happens if I reduce it to:

```typescript
const nilCompute = useNilInvokeCompute()

nilCompute.execute({
  programId: wordle.programId,
  inputBindings: [],
  outputBindings: [],
  computeTimeValues: [],
  valueIds: [],
})
```

Devnet version:

```console
$ nillion-devnet --version
nillion-devnet v0.7.0
Release candidate version: v0.7.0-rc.55
Git commit hash: fc616154927592db88c47048f7723b7acb6d13a8
```

**Comment by tim-hm** [comment]
Is your working branch public so I can try running this locally?

**Reply by a3s7p** [reply]
https://github.com/a3s7p/wordle/tree/sdk-v0.7.0

**Reply by tim-hm** [reply]
Thanks will take a look later today.

**Reply by tim-hm** [reply]
I think its complaining that the user doesn't have permissions?

<img width="1668" alt="image" src="https://github.com/user-attachments/assets/b73e4438-6d0a-4b6c-b587-3d40b2ae24e3">

**Reply by a3s7p** [reply]
Thanks for looking into this!

I have updated the branch. I was giving the hex string of the user ID to `grantCompute` instead of the derived `UserId` object.

However after fixing that it still doesn't work?

<img width="1455" alt="Pasted Graphic" src="https://github.com/user-attachments/assets/53334ac6-28b5-46f5-9d7a-dc2e04c376aa">

<img width="1455" alt="Pasted Graphic 1" src="https://github.com/user-attachments/assets/e01ee28c-ff57-46ab-9666-419e5c551175">

[This](https://github.com/a3s7p/wordle/blob/6dd59e367721caa56493b618d54cd4a48d6221bc/app/components/WordleRowStore.tsx#L32-L35) is how I am setting permissions, is it missing anything?

FWIW I could see Nillion errors & permission issues in the console before.

**Comment by oceans404** [comment]
Here's how I was able to give another example user `123-456-789` permissions in this version using ValuesPermissionsBuilder from client-vms

```
"use client";

import { useNillion, useNilStoreValues } from "@nillion/client-react-hooks";
import { NadaValue, UserId, ValuesPermissionsBuilder } from "@nillion/client-vms";
import { useState, useMemo, type FC } from "react";

export const StoreBlob: FC = () => {
  const mutation = useNilStoreValues();
  const [copied, setCopied] = useState(false);

  let id = "";
  if (mutation.isSuccess) {
    id = mutation.data;
  } else if (mutation.isError) {
    id = mutation.error.message;
  }
  const { client } = useNillion();
  const permissions = useMemo(() => {
    const perms = ValuesPermissionsBuilder.init();
    perms.owner(client.id);
    perms.grantRetrieve(client.id);
    perms.grantRetrieve('123-456-789');
    return perms.build();
  }, [client.id]);

  const options = {
    values: [
      {
        name: "data",
        value: NadaValue.new_secret_blob(
          new TextEncoder().encode("Hello, world!"),
        ),
      },
    ],
    ttl: 1,
    permissions,
  };

  function handleClick(): void {
    mutation.execute(options);
  }
  return (...)
```

**Reply by a3s7p** [reply]
Thanks! still doesn't seem to work for me though — any string value as UserId seems to break it.

<img width="1434" alt="image" src="https://github.com/user-attachments/assets/534cb5e5-de8b-4676-a24e-a603f2e8e99a">

**Comment by tim-hm** [comment]
Currently the UserId can only be created from a UserId type and not directly from a string. So unfortunately `perms.grantRetrieve('123-456-789');` wont work. We're discussing this internally.

**Comment by a3s7p** [comment]
I found a [workaround](https://blog.gets.love/posts/nillion-secret-manager/#debugging-with-sdk-v0-7-0) for the original issue of not seeing the error messages.

**Reply by envoy1084** [reply]
Hey I guess there are some errors while passing store ids to compute, this will be fixed in the next release as per the team.

Meanwhile you can update to `@nillion/client-vms: v0.3.0-rc.3` with local devnet `v0.9.0` for compute with store ids.
